ddtrace/span.py

```diff
@@ -486,14 +488,23 @@ def pprint(self):
             ("resource", self.resource),
             ("type", self.span_type),
             ("start", self.start),
-            ("end", "" if not self.duration else self.start + self.duration),
-            ("duration", "%fs" % (self.duration or 0)),
+            ("end", None if not self.duration else self.start + self.duration),
+            ("duration", self.duration),
             ("error", self.error),
-            ("tags", ""),
+            ("tags", dict(sorted(self.meta.items()))),
+            ("metrics", dict(sorted(self.metrics.items()))),
         ]
-
-        lines.extend((" ", "%r:%s" % kv) for kv in sorted(self.meta.items()))
-        return "\n".join("%10s %s" % line for line in lines)
+        return json.dumps(
```

Comment:
Can we just `pprint.pprint` here instead of json? json could fail but I think pprint is pretty robust for handling python objects. This would clean up having to do the `if isinstance(v, six.integer_types) or isinstance(v, float) or isinstance(v, dict) or v is None` check

Reply:
Unfortunately `pprint` will sort the keys of a dict (an option was added with 3.8 to avoid this https://bugs.python.org/issue30670). So instead I decided to still use it but iterate over the attributes, pprint the value and concatenate them together in our code. Tell me what you think!

Reply:
ah interesting! yeah seems like a nice solution to me :+1:

Reply:
have an example of what the new output looks like?

Reply:
Would be worth having an assertion in the test case with the complete output even though it might be a bit ugly.

Reply:
Added in description